test/unified-config.test.js

```diff
@@ -0,0 +1,453 @@
+    ]
+    doChecks(cfg, {file: fullpath, errors});
+  })
+
+  it('should correctly report an error reading a bad config file', function () {
+    writeConfigLiteral('{"i am: bad\n');
+
+    const cfg = guc();
+
+    const errors = [[
+      'Cannot read config file /home/bruce/solarwinds/ao-agent/appoptics-apm: ',
+      '/home/bruce/solarwinds/ao-agent/appoptics-apm.json: Unexpected token \n in JSON at position 11'
+    ].join('')];
+    doChecks(cfg, {errors});
+  })
+
+  //
+  // env vars
+  //
+  it('should warn about unknown APPOPTICS_ environment variables', function () {
+    process.env.APPOPTICS_YINYIN_ROCKS = 'truth';
+
+    const cfg = guc();
+
+    const unusedEnvVars = ['APPOPTICS_YINYIN_ROCKS=truth'];
+    doChecks(cfg, {unusedEnvVars});
+  })
+
+  it('should not set use environment variable values', function () {
+    process.env.APPOPTICS_EC2_METADATA_TIMEOUT = 'xyzzy';
+
+    const cfg = guc();
+
+    const warnings = ['invalid environment variable value APPOPTICS_EC2_METADATA_TIMEOUT=xyzzy'];
+    doChecks(cfg, {warnings});
+  })
+
+  it('should keep valid config file value when there is a bad env var value', function () {
+    const config = {ec2MetadataTimeout: 1000};
+    writeConfigJSON(config);
+    process.env.APPOPTICS_EC2_METADATA_TIMEOUT = 'xyzzy';
+
+    const cfg = guc();
+
+    const warnings = ['invalid environment variable value APPOPTICS_EC2_METADATA_TIMEOUT=xyzzy'];
+    doChecks(cfg, {global: config, warnings});
+  })
+
+  it('should override config file keys when environment variables are present', function () {
+    const config = {
+      enabled: false,
+      hostnameAlias: 'bruce',
+      serviceKey: 'f'.repeat(64),
+      ignoreConflicts: true,
+      domainPrefix: 'fubar',
+    }
+    writeConfigJSON(config);
+    process.env.APPOPTICS_SERVICE_KEY = 'ab'.repeat(32);
+    process.env.APPOPTICS_HOSTNAME_ALIAS = 'macnaughton';
+
+    const cfg = guc();
+
+    config.hostnameAlias = process.env.APPOPTICS_HOSTNAME_ALIAS;
+    config.serviceKey = process.env.APPOPTICS_SERVICE_KEY;
+    doChecks(cfg, {global: config});
+  })
+
+  it('should correctly handle env vars with explicit names', function () {
+    process.env.APPOPTICS_DEBUG_LEVEL = 4;
+    process.env.APPOPTICS_COLLECTOR = 'collector-stg.appoptics.com';
+
+    const cfg = guc();
+
+    const config = {logLevel: 4, endpoint: process.env.APPOPTICS_COLLECTOR};
+    doChecks(cfg, {global: config});
+  })
+
+  it('should map an env var name to a different property name', function () {
+    process.env.APPOPTICS_TRIGGER_TRACE = 'disabled';
+
+    const cfg = guc();
+
+    const config = {triggerTraceEnabled: false};
+    const warnings = ['APPOPTICS_TRIGGER_TRACE is deprecated; it will be invalid in the future'];
+    doChecks(cfg, {global: config, warnings});
+  })
+
+  //
+  // probes
+  //
+  it('should set probe values correctly', function () {
+    const config = {probes: {fs: {enabled: false, bruce: 'says hello'}}};
```

Comment:
why does it not warn about invalid config name?

Reply:
i don't follow - why should it warn about an invalid config name?